src/libcommonserver/io/iohelper.cpp

```diff
@@ -598,6 +603,18 @@ bool IoHelper::checkIfPathExists(const SyncPath &path, bool &exists, IoError &io
         ioError = IoError::Success;
         return true;
     }
+#ifdef _WIN32 // TODO: Remove this block when we are migrating the release process to Visual Studio 2022
+    // Prior to Visual Studio 2022, std::filesystem::symlink_status would return a missleading InvalidArgument if the path is
+    // found but located on a FAT32 disk. If the file is not found, it works as expected. This behavior is fixed when compiling
+    // with VS2022. https://developercommunity.visualstudio.com/t/std::filesystem::is_symlink-is-broken-on/1638272
+    if (ioError == IoError::InvalidArgument) {
+        if (Utility::isNtfs(path.parent_path()) ||
+            path.string().length() > CommonUtility::maxPathLength()) { // These are the normal cases for a InvalidArgument
```

Comment:
```suggestion
            path.string().length() > CommonUtility::maxPathLength()) { // These are the normal cases for an InvalidArgument
```

Reply:
Are there other natural conditions associated to `InvalidArgument`? The length of file name? Special characters?

Reply:
Here is a revised implementation to avoid overrding `InvalidArgument` errors. [5125d61](https://github.com/Infomaniak/desktop-kDrive/pull/338/commits/5125d61f0b912e0ddf4ec793fa166b1a8b2ffc1e)